src/test/scala/org/broadinstitute/dsde/firecloud/service/LibraryApiServiceSpec.scala

```diff
@@ -237,5 +237,34 @@ class LibraryApiServiceSpec extends BaseServiceSpec with LibraryApiService with
         }
       }
     }
+
+    "when searching for ORSP IDs" - {
+      "GET on " + duosConsentOrspIdPath - {
+        "should return a valid consent for '12345'" in {
+          Get(duosConsentOrspIdPath + "12345") ~> dummyUserIdHeaders("1234") ~> sealRoute(libraryRoutes) ~> check {
+            status should equal(OK)
+            val consent = response.entity.asString.parseJson.convertTo[Consent]
+            consent shouldNot equal(None)
+            consent.name should equal("12345")
+          }
+        }
+        "should return a Bad Request error on 'unapproved'" in {
+          Get(duosConsentOrspIdPath + "unapproved") ~> dummyUserIdHeaders("1234") ~> sealRoute(libraryRoutes) ~> check {
+            status should equal(BadRequest)
+          }
+        }
+        "should return a Not Found error on known 'missing'" in {
+          Get(duosConsentOrspIdPath + "missing") ~> dummyUserIdHeaders("1234") ~> sealRoute(libraryRoutes) ~> check {
+            status should equal(NotFound)
+          }
+        }
+        "should return a Not Found error on unknown missing" in {
+          val orspId = randomStringFromCharList(10, ('a' to 'z') ++ ('A' to 'Z') ++ ('0' to '9'))
+          Get(duosConsentOrspIdPath + orspId) ~> dummyUserIdHeaders("1234") ~> sealRoute(libraryRoutes) ~> check {
+            status should equal(NotFound)
+          }
+        }
+      }
+    }
```

Comment:
These tests are filling the logs with exceptions - I need to figure out a way to clean that up.